data_structures/stacks/postfix_evaluation.py

```diff
@@ -17,52 +24,197 @@
         Result =  59
 """
 
-import operator as op
+
+def get_number(data: str) -> tuple[bool, int] | tuple[bool, float] | tuple[bool, str]:
+    """
+    Converts the given data to appropriate number if it is indeed a number, else returns
+    the data as it is with a False flag. This function also serves as a check of whether
+    the input is a number or not.
+
+    Parameters
+    ----------
+    data : str
+        The data which needs to be converted to the appropriate number
+
+    Returns
+    -------
+    bool,  int or float
+        Returns a tuple of (a, b) where 'a' is True if data is indeed a number (integer
+        or numeric) and 'b' is either an integer of a floating point number.
+        If 'a' is False, then b is 'data'
+    """
+    int_val = 0
+    float_val = 0.0
+    try:
+        int_val = int(data)
+        return True, int_val
+    except ValueError:
+        try:
+            float_val = float(data)
+            return True, float_val
+        except ValueError:
+            return False, data
+
+
+def is_operator(data: str) -> bool:
+    """
+    Checks whether a given input is one of the valid operators or not.
+    Valid operators being '-', '+', '*', '^' and '/'.
+
+    Parameters
+    ----------
+    data : str
+        The value that needs to be checked for operator
+
+    Returns
+    -------
+    bool
+        True if data is an operator else False.
+    """
+    return data in ["-", "+", "*", "^", "/"]
 
 
-def solve(post_fix):
+def evaluate(post_fix: list, verbose: bool = False) -> int | float | str | None:
+    """
+    Function that evaluates postfix expression using a stack.
+    >>> evaluate(["2", "1", "+", "3", "*"])
+    9
+    >>> evaluate(["4", "13", "5", "/", "+"])
+    6.6
+    >>> evaluate(["2", "-", "3", "+"])
+    1
+    >>> evaluate([])
+    0
+
+    Parameters
+    ----------
+    post_fix : list
+        The postfix expression tokenized into operators and operands and stored as a
+        python list
+
+    verbose : bool
+        Display stack contents while evaluating the expression if verbose is True
+
+    Returns
+    -------
+    int
+        The evaluated value
+    """
     stack = []
-    div = lambda x, y: int(x / y)  # noqa: E731 integer division operation
     opr = {
-        "^": op.pow,
-        "*": op.mul,
-        "/": div,
-        "+": op.add,
-        "-": op.sub,
+        "^": lambda p, q: p**q,
+        "*": lambda p, q: p * q,
+        "/": lambda p, q: p / q,
+        "+": lambda p, q: p + q,
+        "-": lambda p, q: p - q,
     }  # operators & their respective operation
+    if len(post_fix) > 0:
+        if verbose:
+            # print table header
+            print("Symbol".center(8), "Action".center(12), "Stack", sep=" | ")
+            print("-" * (30 + len(post_fix)))
```

Comment:
Negate the if-statement to return early. That way you can avoid nesting the code as much.
```suggestion
    if len(post_fix) == 0:
        return 0
    if verbose:
        # print table header
        print("Symbol".center(8), "Action".center(12), "Stack", sep=" | ")
        print("-" * (30 + len(post_fix)))
```